app/plugin.go

```diff
@@ -257,9 +257,7 @@ func (a *App) SyncPlugins() *model.AppError {
 }
 
```

Comment:
The amount of unnecessary lock statements bugs me and will make it hard to debug this in the future. Would you mind clearing them up in this PR? They are:
- https://github.com/mattermost/mattermost-server/blob/master/app/plugin.go#L46-L48
- https://github.com/mattermost/mattermost-server/blob/master/app/plugin.go#L127-L129